### PR TITLE
Fix changes requested in Pr #91

### PR DESCRIPTION
**Pull request by marekkawalski**
Fix changes requested in Pr #91

1. Add missing error handling
2. Add named return values to `unc UnpackShotMsg(message io.Reader) (model.Shot, error)`
3. Replace `if err != nil {
				assert.FailNow(t, err.Error())
			}` with `assert.Nil(t, err)`
4. Add test cases for `func TestGameSubGoal(t *testing.T)`

### CODE REVIEW

File: backend/internal/controller/adapter/shot_message_test.go [hunk 1]
```diff
@@ -63,9 +63,8 @@ func TestUnpackShotMsg(t *testing.T) {
 		t.Run(tt.name, func(t *testing.T) {
 			var rawSlice []json.RawMessage
 			jsn, err := json.Marshal(tt.msgIn)
-			if err != nil {
-				assert.FailNow(t, err.Error())
-			}
+
+			assert.Nil(t, err)
```
Comment (MarekMichali) [inline backend/internal/controller/adapter/shot_message_test.go:67]: Is this a good idea? In the case of unsuccessful marshalling there is probably no point in performing further tests.

File: backend/internal/controller/adapter/shot_message_test.go [hunk 2]
```diff
@@ -75,9 +74,8 @@ func TestUnpackShotMsg(t *testing.T) {
 				Params: rawSlice,
 			}
 			shotJSON, err := json.Marshal(tableShot)
-			if err != nil {
-				assert.FailNow(t, err.Error())
-			}
+
+			assert.Nil(t, err)
```
Comment (MarekMichali) [inline backend/internal/controller/adapter/shot_message_test.go:78]: Is this a good idea? In the case of unsuccessful marshalling there is probably no point in performing further tests.

File: backend/internal/controller/adapter/shot_message_test.go (hunk 1 above)
Comment (Pichi00) [inline backend/internal/controller/adapter/shot_message_test.go:67]:
```suggestion
			if err != nil {
				assert.FailNow(t, err.Error())
			}
```
I would revert this change. Further tests are dependent on Marshal() method. If this method fails I don't see a point in checking the rest of the tests.

Comment (VOID404) [reply backend/internal/controller/adapter/shot_message_test.go:67]: Use [`require.Nil`](https://pkg.go.dev/github.com/stretchr/testify@v1.8.0/require#Nil)

File: backend/internal/controller/adapter/shot_message_test.go (hunk 2 above)
Comment (VOID404) [reply backend/internal/controller/adapter/shot_message_test.go:78]: Use [`require.Nil`](https://pkg.go.dev/github.com/stretchr/testify@v1.8.0/require#Nil)

File: backend/internal/controller/adapter/shot_message_test.go (hunk 1 above)
Comment (VOID404) [reply backend/internal/controller/adapter/shot_message_test.go:67]: Use [`require.Nil`](https://pkg.go.dev/github.com/stretchr/testify@v1.8.0/require#Nil)

File: backend/internal/controller/adapter/shot_message_test.go (hunk 2 above)
Comment (Pichi00) [reply backend/internal/controller/adapter/shot_message_test.go]: Done

File: backend/internal/controller/adapter/shot_message_test.go (hunk 1 above)
Comment (Pichi00) [reply backend/internal/controller/adapter/shot_message_test.go]: Done

Comment (Pichi00) [reply backend/internal/controller/adapter/shot_message_test.go]: Done

**Review by MarekMichali — APPROVED** [review]

**Review by MichalKalke — APPROVED** [review]

**Pull request merged**